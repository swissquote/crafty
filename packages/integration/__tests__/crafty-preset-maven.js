const test = require("ava");
const path = require("path");
const testUtils = require("../utils");

test.serial("Fails if no pom is found", async t => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-maven/missing-pom"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  t.is(result.status, 0);
  t.truthy(
    result.stdall.includes(
      "Could not define destination using maven, falling back to default."
    )
  );
  t.truthy(result.stdall.includes("No pom.xml found"));
  t.snapshot(result);

  t.truthy(testUtils.exists(cwd, "dist/js/myBundle.min.js"));
});

test.serial("Places files in target of a webapp", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-maven/webapp");

  const result = await testUtils.run(["run", "default"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);

  t.truthy(
    testUtils.exists(
      cwd,
      "target/my-app-1.0.0-SNAPSHOT/resources/js/myBundle.min.js"
    )
  );
});

test.serial("Reads env. var before pom.xml", async t => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-maven/env-override"
  );

  const result = await testUtils.run(["run", "default"], cwd, {
    env: { TARGET_BASEDIR: path.join(cwd, "target/some_basedir") }
  });

  t.snapshot(result);
  t.is(result.status, 0);

  t.truthy(
    testUtils.exists(cwd, "target/some_basedir/resources/js/myBundle.min.js")
  );
});

test.serial(
  "Places files in target of a webapp from within a subfolder",
  async t => {
    const cwd = await testUtils.getCleanFixtures(
      "crafty-preset-maven/subfolder/src/main/frontend"
    );

    const result = await testUtils.run(["run", "default"], cwd);

    t.snapshot(result);
    t.is(result.status, 0);

    t.truthy(
      testUtils.exists(
        cwd,
        "../../../target/my-app-1.0.0-SNAPSHOT/resources/js/myBundle.min.js"
      )
    );
  }
);

test.serial("Places files in target of a webjar", async t => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-maven/webjar");

  const result = await testUtils.run(["run", "default"], cwd);

  t.snapshot(result);
  t.is(result.status, 0);

  t.truthy(
    testUtils.exists(
      cwd,
      "target/classes/META-INF/resources/webjars/js/myBundle.min.js"
    )
  );
});
