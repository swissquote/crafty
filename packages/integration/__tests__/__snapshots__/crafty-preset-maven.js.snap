// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fails if no pom is found 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
Could not define destination using maven, falling back to default.
Original message:
No pom.xml found
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-maven/missing-pom/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Places files in target of a webapp 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-maven/webapp/target/my-app-1.0.0-SNAPSHOT/resources/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Places files in target of a webapp from within a subfolder 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-maven/subfolder/target/my-app-1.0.0-SNAPSHOT/resources/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Places files in target of a webjar 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-maven/webjar/target/classes/META-INF/resources/webjars/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Reads env. var before pom.xml 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-maven/env-override/target/some_basedir/resources/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
\`resolve/lib/core\` is deprecated; please use \`is-core-module\` directly
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;
